lib/datadog/core/transport/http/config.rb

```diff
@@ -0,0 +1,258 @@
+require_relative 'client'
+require_relative '../../../../ddtrace/transport/http/response'
+require_relative '../../../../ddtrace/transport/http/api/endpoint'
+
+# TODO: Decouple standard transport/http/api/instance
+#
+# Separate classes are needed because transport/http/trace includes
+# Trace::API::Instance which closes over and uses a single spec, which is
+# negotiated as either /v3 or /v4 for the whole API at the spec level, but we
+# need an independent toplevel path at the endpoint level.
+#
+# Separate classes are needed because of `include Trace::API::Instance`.
+#
+# Below should be:
+# require_relative '../../../../ddtrace/transport/http/api/instance'
+require_relative 'api/instance'
+# Below should be:
+# require_relative '../../../../ddtrace/transport/http/api/spec'
+require_relative 'api/spec'
+
+module Datadog
+  module Core
+    module Transport
+      module HTTP
+        # HTTP transport behavior for remote configuration
+        module Config
+          # Response from HTTP transport for remote configuration
+          class Response
+            include Datadog::Transport::HTTP::Response
+            include Core::Transport::Config::Response
+
+            def initialize(http_response, options = {})
+              super(http_response)
+
+              # TODO: these fallbacks should be improved
+              roots = options[:roots] || []
+              targets = options[:targets] || ''
+              target_files = options[:target_files] || []
+              client_configs = options[:client_configs] || []
+
+              raise TypeError.new(Array, roots) unless roots.is_a?(Array)
+
+              @roots = roots.map do |e|
+                raise TypeError.new(String, e) unless e.is_a?(String)
+
+                begin
+                  decoded = Base64.strict_decode64(e) # TODO: unprocessed, don't symbolize_names
+                rescue ArgumentError
+                  raise DecodeError.new(:roots, e)
+                end
+
+                begin
+                  parsed = JSON.parse(decoded)
+                rescue JSON::ParserError
+                  raise ParseError.new(:roots, e)
+                end
+
+                # TODO: perform more processing to validate content. til then, no freeze
+
+                parsed
+              end
+
+              raise TypeError.new(String, targets) unless targets.is_a?(String)
+
+              @targets = begin
+                begin
+                  decoded = Base64.strict_decode64(targets)
+                rescue ArgumentError
+                  raise DecodeError.new(:targets, e)
+                end
+
+                begin
+                  parsed = JSON.parse(decoded) # TODO: unprocessed, don't symbolize_names
+                rescue JSON::ParserError
+                  raise ParseError.new(:targets, e)
+                end
+
+                # TODO: perform more processing to validate content. til then, no freeze
+
+                parsed
+              end
```

Comment:
```suggestion
 raise TypeError.new(String, targets) unless targets.is_a?(String)

begin
  decoded_targets = Base64.strict_decode64(targets)
rescue ArgumentError => e
  raise DecodeError.new(:targets, e)
end

begin
  parsed_targets = JSON.parse(decoded_targets) # TODO: unprocessed, don't symbolize_names
rescue JSON::ParserError => e
  raise ParseError.new(:targets, e)
end

# TODO: perform more processing to validate content. til then, no freeze
@targets = parsed_targets
```

Reply:
Not sure what this suggestion is about, care to elaborate?

Reply:
Removing the nested `begin` blocks